rfc/20251001-eval-plan-apply-architecture.md

```diff
@@ -0,0 +1,1341 @@
+    some cells refer to the results of the expressions in other cells. Just
+    like in a spreadsheet engine's implementation, OpenTofu needs to discover
+    the relationships between those expressions and then evaluate them gradually
+    until they have all been transformed into a final value.
+
+    The expression graph is completely encapsulated inside the _evaluator_,
+    because it's calculated identically in all phases but the rest of the
+    system doesn't actually need all of the source-level detail, so the
+    evaluator produces the next artifact as its primary result.
+
+- The **resource instance graph** boils the expression graph down to just the
+  essential elements that can potentially cause side-effects during the apply
+  phase: resource instances and the provider instances that manage them.
+
+    This form ignores details such as input variables, output values, and
+    the fact that each module has its own separate namespace. If a resource
+    instance configuration refers to a local value which in turn refers to the
+    results from another resource instance then in the resource instance graph
+    the local value is disregarded and we instead model just the dependency
+    between the resource instances.
+
+    The planning phase is driven primarily by the resource instance graph: it
+    visits each resource instance reported by the evaluator and asks its
+    associated provider to produce a plan by comparing the effective
+    configuration with the current state. Based on the provider's answer,
+    the planning phase then chooses a set of specific actions that would need
+    to be taken to bring the remote system closer to the desired state, but
+    instead of performing those actions directly it produces the next artifact
+    for the apply phase to use.
+
+- The **execution graph** expands the pure relationship information from the
+  resource instance graph to a graph of specific actions that can cause
+  externally-visible side effects, making sure that those actions happen in
+  a suitable order that respects the dependencies from the resource instance
+  graph and respects OpenTofu's own mechanical constraints.
+
+    For example, if the provider's proposed plan for a resource instance
+    was to replace the existing object with a new one matching the latest
+    configuration then the _execution graph_ must capture the individual steps
+    of destroying the previous object and creating the new object as separate
+    nodes, ordering them appropriately based on the `create_before_destroy`
+    settings in the resource configurations.
+
+    The job of the apply phase is to perform all of the operations described
+    in the execution graph, while ensuring that the dependencies between them
+    are correctly respected, and that other constraints like OpenTofu's
+    configurable concurrency limit are respected.
+
+Both of these are valid lenses to view the proposed new architecture through.
```

Comment:
There are three items here, what do you mean by both?  It seems like the paragraph here and below are referring to only two items?